### DISCUSSION TITLE
Actions and Apps Can't Request Review From A Team

### DISCUSSION THREAD

**Discussion by Kurt-von-Laven** [community/community#66049] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

This is a summary of renovatebot/renovate#24226, where this issue was originally reported.

[Forking Renovate](https://github.com/apps/forking-renovate) receives `422 (Unprocessable Entity), Validation Failed: Could not resolve to a node with the global id of 'T_kwDOBKjlfs4AggHY'.` when attempting to request review of a team. Here are the relevant logs:

<details><summary>Forking Renovate Logs</summary>

```
DEBUG: Adding reviewers 'team:dependency-management' to #191 (branch="renovate/frnmst-md-toc-8.x")
DEBUG: POST https://api.github.com/repos/ScribeMD/template-repo/pulls/191/requested_reviewers = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=422 retryCount=0, duration=199) (branch="renovate/frnmst-md-toc-8.x")
DEBUG: 422 Error thrown from GitHub (branch="renovate/frnmst-md-toc-8.x")
{
  "err": {
    "name": "HTTPError",
    "code": "ERR_NON_2XX_3XX_RESPONSE",
    "timings": {
      "start": 1693798732147,
      "socket": 1693798732147,
      "lookup": 1693798732148,
      "connect": 1693798732150,
      "secureConnect": 1693798732154,
      "upload": 1693798732154,
      "response": 1693798732346,
      "end": 1693798732346,
      "phases": {
        "wait": 0,
        "dns": 1,
        "tcp": 2,
        "tls": 4,
        "request": 0,
        "firstByte": 192,
        "download": 0,
        "total": 199
      }
    },
    "message": "Response code 422 (Unprocessable Entity)",
    "stack": "HTTPError: Response code 422 (Unprocessable Entity)\n    at Request.<anonymous> (/opt/containerbase/tools/renovate/36.78.8/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
    "options": {
      "headers": {
        "user-agent": "RenovateBot/36.78.8 (https://github.com/renovatebot/renovate)",
        "accept": "application/json, application/vnd.github.machine-man-preview+json",
        "authorization": "***********",
        "content-type": "application/json",
        "content-length": "59",
        "accept-encoding": "gzip, deflate, br"
      },
      "url": "https://api.github.com/repos/ScribeMD/template-repo/pulls/191/requested_reviewers",
      "hostType": "github",
      "username": "",
      "password": "",
      "method": "POST",
      "http2": false
    },
    "response": {
      "statusCode": 422,
      "statusMessage": "Unprocessable Entity",
      "body": {
        "message": "Validation Failed",
        "errors": [
          "Could not resolve to a node with the global id of 'T_kwDOBKjlfs4AggHY'."
        ],
        "documentation_url": "https://docs.github.com/rest/pulls/review-requests#request-reviewers-for-a-pull-request"
      },
      "headers": {
        "server": "GitHub.com",
        "date": "Mon, 04 Sep 2023 03:38:52 GMT",
        "content-type": "application/json; charset=utf-8",
        "content-length": "226",
        "x-github-media-type": "github.v3; param=machine-man-preview",
        "x-accepted-github-permissions": "pull_requests=write",
        "x-github-api-version-selected": "2022-11-28",
        "x-ratelimit-limit": "5000",
        "x-ratelimit-remaining": "4981",
        "x-ratelimit-reset": "1693801152",
        "x-ratelimit-used": "19",
        "x-ratelimit-resource": "core",
        "access-control-expose-headers": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset",
        "access-control-allow-origin": "*",
        "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
        "x-frame-options": "deny",
        "x-content-type-options": "nosniff",
        "x-xss-protection": "0",
        "referrer-policy": "origin-when-cross-origin, strict-origin-when-cross-origin",
        "content-security-policy": "default-src 'none'",
        "vary": "Accept-Encoding, Accept, X-Requested-With",
        "x-github-request-id": "BED1:87FB:187D4C1:327BDF8:64F5514C",
        "connection": "close"
      },
      "httpVersion": "1.1",
      "retryCount": 0
    }
  }
}

WARN: Failed to assign reviewer (branch="renovate/frnmst-md-toc-8.x")
{
  "err": {
    "hostType": "github",
    "err": {
      "name": "HTTPError",
      "code": "ERR_NON_2XX_3XX_RESPONSE",
      "timings": {
        "start": 1693798732147,
        "socket": 1693798732147,
        "lookup": 1693798732148,
        "connect": 1693798732150,
        "secureConnect": 1693798732154,
        "upload": 1693798732154,
        "response": 1693798732346,
        "end": 1693798732346,
        "phases": {
          "wait": 0,
          "dns": 1,
          "tcp": 2,
          "tls": 4,
          "request": 0,
          "firstByte": 192,
          "download": 0,
          "total": 199
        }
      },
      "message": "Response code 422 (Unprocessable Entity)",
      "stack": "HTTPError: Response code 422 (Unprocessable Entity)\n    at Request.<anonymous> (/opt/containerbase/tools/renovate/36.78.8/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
      "options": {
        "headers": {
          "user-agent": "RenovateBot/36.78.8 (https://github.com/renovatebot/renovate)",
          "accept": "application/json, application/vnd.github.machine-man-preview+json",
          "authorization": "***********",
          "content-type": "application/json",
          "content-length": "59",
          "accept-encoding": "gzip, deflate, br"
        },
        "url": "https://api.github.com/repos/ScribeMD/template-repo/pulls/191/requested_reviewers",
        "hostType": "github",
        "username": "",
        "password": "",
        "method": "POST",
        "http2": false
      },
      "response": {
        "statusCode": 422,
        "statusMessage": "Unprocessable Entity",
        "body": {
          "message": "Validation Failed",
          "errors": [
            "Could not resolve to a node with the global id of 'T_kwDOBKjlfs4AggHY'."
          ],
          "documentation_url": "https://docs.github.com/rest/pulls/review-requests#request-reviewers-for-a-pull-request"
        },
        "headers": {
          "server": "GitHub.com",
          "date": "Mon, 04 Sep 2023 03:38:52 GMT",
          "content-type": "application/json; charset=utf-8",
          "content-length": "226",
          "x-github-media-type": "github.v3; param=machine-man-preview",
          "x-accepted-github-permissions": "pull_requests=write",
          "x-github-api-version-selected": "2022-11-28",
          "x-ratelimit-limit": "5000",
          "x-ratelimit-remaining": "4981",
          "x-ratelimit-reset": "1693801152",
          "x-ratelimit-used": "19",
          "x-ratelimit-resource": "core",
          "access-control-expose-headers": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset",
          "access-control-allow-origin": "*",
          "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
          "x-frame-options": "deny",
          "x-content-type-options": "nosniff",
          "x-xss-protection": "0",
          "referrer-policy": "origin-when-cross-origin, strict-origin-when-cross-origin",
          "content-security-policy": "default-src 'none'",
          "vary": "Accept-Encoding, Accept, X-Requested-With",
          "x-github-request-id": "BED1:87FB:187D4C1:327BDF8:64F5514C",
          "connection": "close"
        },
        "httpVersion": "1.1",
        "retryCount": 0
      }
    },
    "message": "external-host-error",
    "stack": "Error: external-host-error\n    at handleGotError (/opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/util/http/github.ts:149:12)\n    at GithubHttp.request (/opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/util/http/github.ts:371:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at GithubHttp.requestJson (/opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/util/http/index.ts:289:17)\n    at Proxy.addReviewers (/opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/modules/platform/github/index.ts:1308:5)\n    at addParticipants (/opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/workers/repository/update/pr/participants.ts:88:11)\n    at ensurePr (/opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/workers/repository/update/pr/index.ts:502:9)\n    at processBranch (/opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/workers/repository/update/branch/index.ts:750:28)\n    at writeUpdates (/opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/workers/repository/process/write.ts:159:17)\n    at update (/opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/workers/repository/process/extract-update.ts:214:11)\n    at Object.renovateRepository (/opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/workers/repository/index.ts:76:19)\n    at attributes.repository (/opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/workers/global/index.ts:184:11)\n    at start (/opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/workers/global/index.ts:169:7)\n    at /opt/containerbase/tools/renovate/36.78.8/node_modules/renovate/lib/renovate.ts:18:22"
  }
}

DEBUG: Added reviewers (branch="renovate/frnmst-md-toc-8.x")
{
  "reviewers": [
    "team:dependency-management"
  ]
}
```

</details>

[The official docs for the pertinent REST endpoint](https://docs.github.com/en/rest/pulls/review-requests?apiVersion=2022-11-28#request-reviewers-for-a-pull-request--status-codes) say that a 422 means the user is not a collaborator, which is contradicted by the fact that we are able to request review from the team manually.

I got [the same error message (including the same ID)](https://github.com/ScribeMD/slack-templates/actions/runs/6070948686/job/16468001352) trying to request review from the Dependency Management team from a GitHub Action in [a different repository](https://github.com/ScribeMD/slack-templates) within [the same organization](https://github.com/ScribeMD) even with [`permissions: write-all`](https://github.com/ScribeMD/slack-templates/actions/runs/6070948686/workflow#L6). 

<details><summary>GitHub Actions Logs</summary>

```
Run curl --fail-with-body --silent --show-error --request POST --header "Accept: application/vnd.github+json" --header "Authorization: ***" --header "X-GitHub-Api-Version: 2022-11-28" "$GITHUB_API_URL/repos/$GITHUB_REPOSITORY/pulls/426/requested_reviewers" --data '{"team_reviewers":["dependency-management"]}'
  shell: /usr/bin/bash -e {0}
curl: (22) The requested URL returned error: [4](https://github.com/ScribeMD/slack-templates/actions/runs/6070948686/job/16468001352#step:2:5)22
{
  "message": "Validation Failed",
  "errors": [
    "Could not resolve to a node with the global id of 'T_kwDOBKjlfs4AggHY'."
  ],
  "documentation_url": "https://docs.github.com/rest/pulls/review-requests#request-reviewers-for-a-pull-request"
}
Error: Process completed with exit code 22.
```

</details>

I assume, therefore, that the ID is that of the Dependency Management team. There are some actions that can only be performed with a PAT though for security reasons, and I had no problems requesting review of the Dependency Management team using a fine-grained PAT (Personal Access Token) with only `pull-requests: write` and the mandatory `metadata: read` permission. If [the review request REST endpoint](https://docs.github.com/en/rest/pulls/review-requests?apiVersion=2022-11-28#request-reviewers-for-a-pull-request) is intended to require a PAT, then [the documentation](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-pull-requests) and error message should be corrected to reflect that. Otherwise, the REST endpoint should behave as documented when used by a GitHub App or GitHub Action, and I am happy to support further troubleshooting efforts.

**Comment by LiamMacP** [comment]
Building on this slightly with our own use-case. I am part of an organisation and we have a team that we would like to assign to pull requests through an action call (see below):

```
      - name: Create Pull Request
        uses: actions/github-script@v6
        with:
          script: |
            [... shortened for brevity - creates PR]
            await github.rest.pulls.requestReviewers({
              owner,
              repo,
              pull_number: result.data.number,
              team_reviewers: ['team-name']
            }); 
```

The permissions this action have are: 
- `pull-requests: write`
- `contents: write`
The only implicit one is metadata, which likely pulls the contributors for the repository.

We are finding that we are hitting the same error: `Validation Failed: "Could not resolve to a node with the global id of '[redacted]'."`

Does anyone know why specifying `team_reviewers` here causes this? Why would the permissions granted not allow this?

We cannot find anything within the documentation around [GitHub Actions permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) that talk about specific use cases being blocked or needing a PAT. Utilising a PAT token in a local project works, although this shouldn't be a requirement and isn't appropriate for what feels like a system action.

---

Building on this the `global id` it is trying to find is the node id of the organisation `team-name` that has been specified.

**Comment by bmehta001** [comment]
Still happening in 2026 with default GITHUB_TOKEN. Adding a fresh data point since the thread is old.

Same call shape: `team_reviewers` via `dyladan/component-owners@v0.2.0`, private team in the same org as the repo, `GITHUB_TOKEN` scoped `pull-requests: write`. Failing job: https://github.com/microsoft/cpp_client_telemetry_modules/actions/runs/26686099878/job/78654499889

```
Validation Failed: "Could not resolve to a node with the global id of 'MDQ6VGVhbTMzODc4NTQ='."
```

It seems that the id format isn't really the issue. Mine shows the legacy `MDQ6` form, but the original report here was already a modern `T_kw...` id, and same in https://github.com/octokit/octokit.js/discussions/2659. Same call with a fine-grained PAT (`pull-requests: write`) works against the exact same team, so it's the `GITHUB_TOKEN` code path that's broken, not id resolution. The error message just makes it harder to figure out.

Open 2.5 years now. Any chance of a fix, or at least an accurate message?

**Comment by eijimiyawaki-sys** [comment]
Adding another data point since this is still open in 2026.

We hit the exact same failure from a third-party GitHub App integration (not one we built ourselves) installed on a private organization repository. Before assuming a permissions gap, we explicitly verified the app's installation permissions in Organization Settings → GitHub Apps:


Read access to members, commit statuses, and metadata (Members: Read confirmed present)
Read/write access to pull requests, issues, checks, actions, and repository hooks
The target repository is included in the app's repository access selection


Despite all of the above being granted, calling the "Request reviewers for a pull request" endpoint with team_reviewers set to a valid team slug in the same org still returns:

422 Unprocessable Entity
Validation Failed: "Could not resolve to a node with the global id of '<team-node-id>'."

Individual (user) reviewers work fine through the same integration. This matches what others in this thread have reported: the failure is specific to team_reviewers resolution through GitHub App installation tokens / GITHUB_TOKEN, not a permissions problem on the installing organization's side. Confirming this because "just grant more permissions" seems to be the most common troubleshooting advice we saw elsewhere, and in our case that advice didn't apply — the permission was already there.

Given this has been open since 2023 and is still reproducing in 2026, would appreciate any update on whether this is being tracked internally, or whether the fix is expected to land as "use a PAT for this specific call" being the permanent recommended workaround rather than a bug that will be fixed.
